compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -422,6 +422,7 @@ impl<'p, 'tcx> RustcMatchCheckCtxt<'p, 'tcx> {
             | ty::RawPtr(_)
             | ty::FnDef(_, _)
             | ty::FnPtr(_)
+            | ty::Pat(_, _)
```

Comment:
Do you have plans for pattern types to affect pattern matching anytime soon? I'd have good fun implementing that, do ping me if I can help

Reply:
This implementation is meant to not be special at all, and any extensions will need to go through another T-types MCP: https://github.com/rust-lang/types-team/issues/126#issuecomment-1898792296

Also:
> Going beyond this simple new type variant requires an additional MCP. Any additional attempts at adding complexity to this feature may be halted by just referring to this MCP's text.